cohortextractor/query_language.py

```diff
@@ -1,3 +1,7 @@
+from dataclasses import dataclass
+from typing import Any
```

Comment:
We don't appear to be running any type checking on this project, so I'm not sure using Any (which effectively disables type checking if we _were_ using a checker) is useful here.  Do we need to set types on the dataclass variables?

Reply:
>  Do we need to set types on the dataclass variables?

As far as I understand (and I may be very wrong) the syntax that dataclasses uses require you to specify a type.  For most of these dataclasses, I'm not sure what the range of valid types is.  It would certainly be good to pin these down, but I don't see much point in doing so until we're actually checking the types.  (This module feels like a good candidate for proper type checking.)